components/brave_wallet/renderer/js_solana_provider.cc

```diff
@@ -0,0 +1,790 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/renderer/js_solana_provider.h"
+
+#include <tuple>
+#include <utility>
+
+#include "base/no_destructor.h"
+#include "brave/components/brave_wallet/common/brave_wallet_constants.h"
+#include "brave/components/brave_wallet/common/brave_wallet_response_helpers.h"
+#include "brave/components/brave_wallet/common/solana_utils.h"
+#include "brave/components/brave_wallet/common/web3_provider_constants.h"
+#include "brave/components/brave_wallet/renderer/v8_helper.h"
+#include "brave/components/brave_wallet/resources/grit/brave_wallet_script_generated.h"
+#include "components/grit/brave_components_strings.h"
+#include "content/public/renderer/v8_value_converter.h"
+#include "gin/array_buffer.h"
+#include "gin/handle.h"
+#include "gin/object_template_builder.h"
+#include "third_party/blink/public/common/browser_interface_broker_proxy.h"
+#include "third_party/blink/public/web/web_console_message.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/resource/resource_bundle.h"
+#include "v8/include/v8-microtask-queue.h"
+#include "v8/include/v8-typed-array.h"
+
+namespace brave_wallet {
+
+namespace {
+
+static base::NoDestructor<std::string> g_provider_script("");
+
+std::string LoadDataResource(const int id) {
```

Comment:
As a follow up pls move this to a shared file similar to v8_helper.cc, maybe resoure_helper.cc

Reply:
https://github.com/brave/brave-browser/issues/21944